packages/flutter/test/material/text_field_test.dart

```diff
@@ -8574,6 +8574,36 @@ void main() {
     expect(scrollController.offset, 48.0);
   });
 
+  testWidgets('TextField is visible on the screen when the cursor is not shown', (WidgetTester tester) async {
+    final ScrollController scrollController = ScrollController();
+
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(),
+      home: Scaffold(
+        body: Center(
+          child: ListView(
+            controller: scrollController,
+            children: <Widget>[
+              Container(height: 579), // Push field almost off screen.
+              const TextField(
+                  showCursor: false,
+              ),
+              Container(height: 1000),
+            ],
+          ),
+        ),
+      ),
+    ));
+
+    // Tap the TextField to put the cursor into it and bring it into view.
+    expect(scrollController.offset, 0.0);
+    await tester.tapAt(tester.getTopLeft(find.byType(TextField)));
+    await tester.pumpAndSettle();
+
+    // The ListView has scrolled to keep the TextField visible.
+    expect(scrollController.offset, 50.0);
```

Comment:
Please pay attention to the previous test case(line 8574). There are more scrolling by 2 pixels than set `showCursor` to true. It is because of the different algorithms. I did not try to keep them consistent, because I think this is not particularly important, right?